### PR TITLE
feat: add internationalization (i18n) support (#207)

### LINKED ISSUE #207: Internationalization (i18n) support
## Summary
Add support for translated diagnostic messages.

## Tasks
- [ ] Extract all user-facing strings
- [ ] Create translation framework
- [ ] Add initial translations (Spanish, Chinese, Japanese, etc.)
- [ ] Allow community translations
- [ ] Detect locale from environment

## Acceptance Criteria
- Framework in place
- At least 2 languages supported
- Contribution guide for translators

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

- Add internationalization (i18n) support using `rust-i18n` crate with compile-time YAML translation loading
- Extract ~420 user-facing strings across 4 crates into translation keys with `t!()` macro
- Support 3 locales: English (en), Spanish (es), Chinese Simplified (zh-CN)
- Add `--locale` CLI flag, `AGNIX_LOCALE` env var, and `.agnix.toml` locale field for locale selection
- Add locale detection module with priority chain: CLI flag > env var > config > system > en
- Initialize locale in LSP server from environment and workspace config
- Force English locale for machine-readable output (JSON/SARIF) for CI/CD compatibility
- Add comprehensive translator contribution guide

### Infrastructure
- Added `rust-i18n` 3.1 and `sys-locale` 0.3 workspace dependencies
- Created `locales/` directory with `en.yml`, `es.yml`, `zh-CN.yml` translation files
- Added `i18n!()` macro initialization in agnix-core, agnix-cli, and agnix-lsp crate roots

### Locale Detection
- New `crates/agnix-cli/src/locale.rs` module with normalization, detection, and supported locale checking
- New `crates/agnix-lsp/src/locale.rs` for LSP server locale initialization
- Added `--locale <LOCALE>` and `--list-locales` CLI flags
- Added `locale: Option<String>` field to `LintConfig`

### String Extraction (13 validators + CLI + LSP)
- Replaced all diagnostic messages, suggestions, assumptions, and fix descriptions with `t!()` calls
- Replaced CLI output strings in `main.rs` and `watch.rs`
- Replaced LSP diagnostic mapper "Suggestion:" label
- Config warning messages internationalized

### Documentation
- `docs/TRANSLATING.md` - Comprehensive translator contribution guide
- Updated README.md with i18n usage section
- Updated SPEC.md with locale CLI flags and config field
- Updated CONTRIBUTING.md with translation reference
- Updated CHANGELOG.md with i18n feature entry

## Test Plan

- [x] All 1633 tests pass (`cargo test`)
- [x] Release build succeeds (`cargo build --release`)
- [x] Zero clippy warnings (`cargo clippy -- -D warnings`)
- [x] 8 i18n-specific unit tests (translation loading, fallback, parameterization)
- [x] 14 CLI integration tests (--locale flag, --list-locales, AGNIX_LOCALE env, config locale, priority order, edge cases)
- [x] 7 LSP locale unit tests (normalization, config init, env init)
- [x] 8 locale edge case tests (empty, whitespace, long strings, special chars)
- [x] JSON/SARIF output verified English regardless of locale
- [x] Verify: `cargo run --bin agnix -- . --locale es` produces Spanish output
- [x] Verify: `AGNIX_LOCALE=zh-CN cargo run --bin agnix -- .` produces Chinese output

Closes #207

### CODE REVIEW

File: crates/agnix-cli/src/locale.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+/// Normalize a locale string to match our supported locale codes.
+///
+/// Examples:
+/// - "en_US.UTF-8" -> "en"
+/// - "es_ES" -> "es"
+/// - "zh_CN.UTF-8" -> "zh-CN"
+/// - "zh-Hans" -> "zh-CN"
+fn normalize_locale(locale: &str) -> String {
+    // Strip encoding suffix (e.g., ".UTF-8")
+    let base = locale.split('.').next().unwrap_or(locale);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-cli/src/locale.rs]: Fixed in commit 4b0bf5d. Added `.trim()` at the start of `normalize_locale()` in CLI to handle input like " en ".

File: crates/agnix-lsp/src/locale.rs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    // 4. Fallback
+    "en".to_string()
+}
+
+/// Normalize a locale string to match supported locale codes.
+fn normalize_locale(locale: &str) -> String {
+    // Strip encoding suffix (e.g., ".UTF-8")
+    let base = locale.split('.').next().unwrap_or(locale);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-lsp/src/locale.rs]: Fixed in commit 4b0bf5d. Added `.trim()` at the start of `normalize_locale()` in LSP to handle whitespace input.

File: crates/agnix-core/src/rules/imports.rs [hunk 3]
```diff
@@ -506,12 +504,9 @@ fn validate_markdown_links(
                     link.line,
                     link.column,
                     "REF-002",
-                    format!("{} target not found: {}", link_type, link.url),
+                    t!("rules.ref_002.message", url = format!("{} target: {}", link_type, link.url), resolved = resolved.display().to_string()),
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/rules/imports.rs]: Fixed in commit 4b0bf5d. The translation now receives the correct parameters: `url` (the link URL) and `resolved` (the resolved file path). Removed the incorrect `link_type` formatting.

File: crates/agnix-core/benches/fixtures.rs [hunk 4]
```diff
@@ -355,7 +355,7 @@ fn create_mcp_tool(index: usize) -> String {
   }},
   "requiresApproval": {approval}
 }}"#,
-        approval = index % 2 == 0
+        approval = index.is_multiple_of(2)
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/benches/fixtures.rs]: Fixed in commit 4b0bf5d. Reverted to `index % 2 == 0` to keep changes focused on i18n. The style improvement is unrelated to this PR scope.

File: crates/agnix-cli/src/telemetry/client.rs [hunk 5]
```diff
@@ -73,10 +70,7 @@ impl TelemetryClient {
         if response.status().is_success() {
             Ok(())
         } else {
-            Err(io::Error::new(
-                io::ErrorKind::Other,
-                format!("Telemetry server returned error: {}", response.status()),
-            ))
+            Err(io::Error::other(format!("Telemetry server returned error: {}", response.status())))
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-cli/src/telemetry/client.rs]: Fixed in commit 4b0bf5d. Reverted to `io::Error::new(io::ErrorKind::Other, ...)` to keep changes focused on i18n. The `io::Error::other` shorthand is unrelated to this PR scope.

File: AGENTS.md [hunk 6]
```diff
@@ -24,6 +24,7 @@
 10. **No unnecessary files** - Don't create summary files, plan files, or temp docs unless specifically required
 11. **Never merge without waiting for claude workflow to end successfully** - It might take time, but this is the major quality gate, and most thorough review.
 12. **You MUST follow the flow phases one by one** - If they state to use subagents, tools, or any specific method, you must follow it exactly as described.
+13. **You MUST address all comments and reviews** - If reviewers leave comments, even minor ones, and even if not a requested change, you must address them all before merging. If you disagree, respond in the review comments. Minor comments must still be addressed.
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply AGENTS.md:27]: This is not an issue. Rule #13 was added to main branch in commit 780b5fe (docs: add guideline to address all comments and reviews before merging) on Dec 23, 2024. Both AGENTS.md and CLAUDE.md are correctly identical as enforced by CI tests. No changes made.

File: crates/agnix-lsp/src/locale.rs [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    let lang = base.split(&['_', '-'][..]).next().unwrap_or(base);
+    for &code in SUPPORTED_LOCALES {
+        let code_lang = code.split('-').next().unwrap_or(code);
+        if lang.eq_ignore_ascii_case(code_lang) {
+            return code.to_string();
+        }
+    }
+
+    lang.to_lowercase()
+}
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-lsp/src/locale.rs]: Fixed in commit 4b0bf5d. Created a shared `agnix_core::i18n` module containing `normalize_locale()` and `is_supported()` functions. Both CLI and LSP now import from this shared module, eliminating the duplication. Added comprehensive tests in the new module.

File: crates/agnix-cli/src/locale.rs [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+/// 2. `config_locale` from `.agnix.toml` locale field
+/// 3. Auto-detection (env vars, system locale, fallback to "en")
+pub fn init(cli_locale: Option<&str>, config_locale: Option<&str>) {
+    let explicit = cli_locale.or(config_locale);
+    let locale = if let Some(l) = explicit {
+        let normalized = normalize_locale(l);
+        if is_supported(&normalized) {
+            normalized
+        } else {
+            eprintln!("Warning: unsupported locale '{}', falling back to 'en'", l);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-cli/src/locale.rs]: Fixed in commit 93265d1. Replaced hardcoded English warning with `t!("cli.locale_unsupported", locale = l)` to use the translation key.

File: docs/TRANSLATING.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+3. **Register the locale** in `crates/agnix-cli/src/locale.rs`:
+   ```rust
+   pub const SUPPORTED_LOCALES: &[(&str, &str)] = &[
+       ("en", "English"),
+       ("es", "Spanish / Espanol"),
+       ("zh-CN", "Chinese Simplified / Zhongwen"),
+       // Add your locale here:
+       ("fr", "French / Francais"),
+   ];
+   ```
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply docs/TRANSLATING.md]: Fixed in commit 93265d1. Updated TRANSLATING.md to document that SUPPORTED_LOCALES is in `crates/agnix-core/src/i18n.rs` and SUPPORTED_LOCALES_DISPLAY is in `crates/agnix-cli/src/locale.rs`, with clear instructions for both.

File: locales/es.yml [hunk 10]
```diff
@@ -0,0 +1,494 @@
+# ===========================================================================
+# Reglas - Mensajes de diagnostico del validador, sugerencias, supuestos, correcciones
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply locales/es.yml]: Fixed in commit 93265d1. All occurrences of "diagnostico" have been replaced with "diagnóstico" (with the accent) throughout the Spanish translation file.
